backend/de.metas.contracts/src/main/java/de/metas/contracts/callorder/price/CallOrderPricingRule.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contracts.callorder.price;
+
+import ch.qos.logback.classic.Level;
+import de.metas.contracts.FlatrateTermId;
+import de.metas.contracts.IFlatrateDAO;
+import de.metas.contracts.callorder.CallOrderContractService;
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.currency.Currency;
+import de.metas.currency.CurrencyRepository;
+import de.metas.logging.LogManager;
+import de.metas.money.CurrencyId;
+import de.metas.money.Money;
+import de.metas.pricing.IPricingContext;
+import de.metas.pricing.IPricingResult;
+import de.metas.pricing.rules.IPricingRule;
+import de.metas.product.ProductId;
+import de.metas.product.ProductPrice;
+import de.metas.tax.api.TaxCategoryId;
+import de.metas.uom.IUOMConversionBL;
+import de.metas.uom.UomId;
+import de.metas.util.ILoggable;
+import de.metas.util.Loggables;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_C_OrderLine;
+import org.slf4j.Logger;
+
+public class CallOrderPricingRule implements IPricingRule
```

Comment:
i get that the code in this class is different from `backend/de.metas.contracts/src/main/java/de/metas/contracts/callorder/price/CallOrderContractLinePricingRule.java`,
but please add some javadoc do describe when each new pricing rule is about